added function term map and function triples map (#14)
kg-construct/rml-core: FNML/definitions.md

  An <a>execution</a> has as result the value of the output of the function.
  For example, `sum(2, 4)` is an execution.
  The value of the output is known after the function is executed, and should in this case be the integer `6`.
- A <dfn>function triples map</dfn> is a specific type of [triples map](https://rml.io/specs/rml/#triples-map): it is used to describe an [=execution=] using [=RML=].

>   As opposed to a [triples map](https://rml.io/specs/rml/#triples-map), a [=function triples map=] has optional [logical source](https://rml.io/specs/rml/#logical-source) and [subject map](https://rml.io/specs/rml/#subject-map).

Reviewer: ```suggestion
  As opposed to a [triples map](https://rml.io/specs/rml/#triples-map), specifying the [logical source](https://rml.io/specs/rml/#logical-source) and [subject map](https://rml.io/specs/rml/#subject-map) of a [=function triples map=] is optional.
```

>   When no logical source is given, the logical source of the triples map referring to the function triples map is used.

Reviewer: ```suggestion
  When no logical source is specified, the logical source of the triples map referring to the function triples map is used.
```

>   When no subject map is given, a blank node as subject is generated.

Reviewer: ```suggestion
  When no subject map is specified, a blank node is generated as subject.
```

Reviewer: Should we add that the blank node should have a unique ID? Or do we leave this up to the implementation?

Author: I think they indeed should have a unique ID, otherwise provenance info generation will be problematic

Author: Added it in the MR

- A <dfn>function triples map</dfn> is a specific type of [=triples map=]: it is used to describe an [=execution=] using [=RML=], but the generated execution triples are not part of the output.
  The output of a function triples map is a function execution result, an internal set of triples which can be used by one or more [=output triples map=].
- A <dfn>function term map</dfn> is a specific type of [term map](https://rml.io/specs/rml/#term-map): it is referenced from an [=output triples map=], generating an [RDF term](https://rml.io/specs/rml/#rdf-term) from the execution result of a [=function triples map=].

<p class="note" data-format="markdown">
The [logical source](https://rml.io/specs/rml/#logical-source) and [subject map](https://rml.io/specs/rml/#subject-map) of a [=function triples map=] should be specified, conform with the [=triples map=] definition.
However, for brevity, following two exceptions are allowed.
When no subject map is specified, a blank node is generated as subject of the [=execution=].
When no logical source is specified, the logical source of the [=output triples map=] referring to the [=function triples map=] is used.

> If in that case multiple logical sources can be linked to the function triples map, the mapping is invalid.

Reviewer: This seems limiting. This means I cannot define a function triples map (with no logical source) and reuse it in function term maps on different triples maps.
It would be more reusable to define that if no logical source is specified the logical source defaults to the logical source of the triples map that is being executed.

That being said. I still struggle to see the use case for defining a logical source on a function triples map at all. I'll raise a new issue to discuss this.
